doc/publishing.md

```diff
@@ -0,0 +1,96 @@
+Publishing the project to Sonatype and Maven Central
+=====================================================
+
+The project relies on some commonly-used sbt plugins to produce artifacts and publish them on public repositories
+
+- [sbt-pgp](https://www.scala-sbt.org/sbt-pgp/) is used to sign the artifacts and publish them
+- [sbt-sonatype](https://github.com/xerial/sbt-sonatype) add some command-line tooling to easily interact with sonatype
+
+Instructions are detailed in the [sbt documentation](https://www.scala-sbt.org/release/docs/Using-Sonatype.html) and will be summarized here with specifics of this project.
+
+## Signing
+
+We publish signed artifacts using the sbt-pgp plugin.
+
+<big>_You need to have the gnu-pgp tool installed and visible from the shell `path` to properly sign the artifacts._</big>
+
+If `gpg` is not in the path, you need to add the following line to the `build.sbt` (or better yet, to a global custom `xxx.sbt` file):
+```scala
+gpgCommand := "/mypath/to/gpg"
+```
+
+In case you're unable to have `gpg` running locally, you can _disable the plugin integration_ with the command line tool and use the plugin's internal _Bouncy Castle_ implementation.
+Change the `build.sbt` to do that:
+```scala
+useGpg := false
+```
+
+## Credentials
+
+The `sbt-sonatype` plugin is already availble the project
+
+You need to tell where the sonatype credentials are stored. To do so, add a global `~/.sbt/1.0/sonatype.sbt` build definition with the following:
+```scala
+credentials += Credentials(Path.userHome / ".sbt" / "sonatype-cryptonomic.credentials")
+```
+
+This instructs the plugin to look for your Sonatype credentials in the custom `~/.sbt/sonatype-cryptonomic.credentials` file, that will contain the following:
```

Comment:
I was acting considering this a safe choice for our integration environment, but maybe we should go with some environment variable?

Is a local file with sonatype credentials too much of a risk?